docs/source/config_file.rst

```diff
@@ -65,234 +65,346 @@ unfortunate, and is subject to change in future versions.
 
-- ``warn_redundant_casts`` (Boolean, default False) warns about
-  casting an expression to its inferred type.
+If you place this file at the root of your repo and run mypy, mypy will:
 
-- ``warn_unused_configs`` (Boolean, default False) warns about
-  per-module sections in the config file that didn't match any
-  files processed in the current run.
+1.  Type-check your entire project assuming it will be run using Python 2.7.
+    (This is equivalent to using the ``--python-version 2.7`` or ``--2`` flag).
 
-- ``scripts_are_modules`` (Boolean, default False) makes script ``x``
-  become module ``x`` instead of ``__main__``.  This is useful when
-  checking multiple scripts in a single run.
+2.  Report an error whenever a function returns a value that is inferred
+    to have type ``Any``.
 
-- ``verbosity`` (integer, default 0) controls how much debug output
-  will be generated.  Higher numbers are more verbose.
+3.  Report any config options that are unused by mypy. (This will help us catch typos
+    when making changes to our config file).
 
-- ``pdb`` (Boolean, default False) invokes pdb on fatal error.
+4.  Selectively disallow untyped function definitions only within the ``mycode.foo``
+    package -- that is, only for function definitions defined in the
+    ``mycode/foo`` directory.
+    
+5.  Selectively *disable* the "function is returning any" warnings within only
+    ``mycode.bar``. This overrides the global default.
 
-- ``show_traceback`` (Boolean, default False) shows traceback on fatal
-  error.
+6.  Suppress any error messages generated when your codebase tries importing the module 
+    ``somelibrary``. (We assume here that ``somelibrary`` is some 3rd party
+    library missing type hints).
 
-- ``dump_type_stats`` (Boolean, default False) dumps stats about type
-  definitions.
 
-- ``dump_inference_stats`` (Boolean, default False) dumps stats about
-  type inference.
+.. _per-module-flags:
 
-- ``incremental`` (Boolean, default True) enables :ref:`incremental
-  mode <incremental>`.
+Per-module and global options
+*****************************
 
-- ``cache_dir`` (string, default ``.mypy_cache``) stores module cache
-  info in the given folder in :ref:`incremental mode <incremental>`.
-  The cache is only read in incremental mode, but it is always written
-  unless the value is set to ``/dev/null`` (UNIX) or ``nul``
-  (Windows).
+The following config options may be set either globally (in the ``[mypy]`` section)
+or on a per-module basis (in sections like ``[mypy-foo.bar]``).
 
-- ``quick_and_dirty`` (Boolean, default False) enables :ref:`quick
-  mode <quick-mode>`.
+If you set an option both globally and for a specific module, the module configuration
+options take precedence. This lets you set global defaults and override them on a
+module-by-module basis. If multiple pattern sections match a module, the options from the
+most specific section are used where they disagree. See above for more details.
```

Comment:
Add a deep link to the section referred to? (`When options conflict, the precedence order for the configuration sections is:`)

Reply:
Done!